mainwindow.cpp

```diff
@@ -0,0 +1,197 @@
+#include "mainwindow.h"
+#include "ui_mainwindow.h"
+#include <QDebug>
+#include <QSerialPortInfo>
+#include <QSerialPort>
+#include <QString>
+#include <QComboBox>
+#include <QTimer>
+#include <QColorDialog>
+#include <QIcon>
+#include <QPixmap>
+#include <QPainter>
+
+// Constructor
+MainWindow::MainWindow(QWidget *parent)
+    : QMainWindow(parent)
+    , ui(new Ui::MainWindow)
+{
+    ui->setupUi(this);
+    setWindowTitle("LED Colour Adjuster");
+
+    serialPort = new QSerialPort(this);
+
+    // Timer to update available COM ports every second
+    QTimer *timer = new QTimer(this);
+    connect(timer, &QTimer::timeout, this, &MainWindow::updateAvailablePorts);
+    timer->start(1000);
+
+    updateControlLockState();
+}
+
+// Destructor
+MainWindow::~MainWindow()
+{
+    delete ui;
+}
+
+// Create command for serial communication
+QString MainWindow::createCommand(const QString &code, int value)

```

Comment:
The first argument can also be passed as `int`. And please rename it to "address"